phoenix-core/src/it/java/org/apache/phoenix/end2end/CDCQueryIT.java

```diff
@@ -701,8 +698,14 @@ public void testCDCIndexBuildAndVerification() throws Exception {
             }
 
             long startTS = System.currentTimeMillis();
-            generateChanges(startTS, tenantids, tableFullName, tableFullName, COMMIT_SUCCESS, null);
-            EnvironmentEdgeManager.reset();
+            List<ChangeRow> changes = generateChanges(startTS, tenantids, tableFullName,
+                    tableFullName, COMMIT_SUCCESS, null, 0);
+            // Make sure the timestamp of the mutations are not in the future
```

Comment:
Why not start with a startTS that is in the past instead of sleeping below?

Reply:
If you do that then whenever we change the code in generateChanges, then we need to change the code here. Also, it is not straightforward to see how much time spent in generateChanges and thus it is not easy for a reader to see if we use the correct timestamp in the past. Further, I am sure if there will be some side effect if we rewind the time back.

Reply:
> Further, I am sure if there will be some side effect if we rewind the time back.

Tried it in an IT, some future calls get stuck in sleep, I think sleep duration is getting miscalculated somewhere.